### PR TITLE
Refactoring of media player and queue

### LINKED ISSUE #102: Player Queue

### PR DESCRIPTION
**Pull request by SimonSimCity**
Fixes #102 

Refactoring of the media player and redesign of the queue. It currently supports:

* Fix click-area to be full width of dropdown for tracks (142febcbc30e2b0f29b3dfc8e04062399b5aad17)
* Add basic queue to UI (0e12510e78eb07c2d0e691c5cf0175bc590f7330)
* Adding tracks as "play next"
* Adding tracks as "add to queue"
* Continue playing when queue has finished playing and track is added to queue
* Loading indicator if track-data is loading
* [Shuffling (as YT-Music)](https://github.com/bcc-code/bmm-web/issues/102#issuecomment-1576504622)
* Replacing the queue when clicking on an item to play (also resets shuffle-option)
* Play a track from the queue
* Report current playing position and implement seeking
* Added tests

Things still to implement:
* [ ] #176
* [ ] Repeat (single, queue)
* [ ] Autoplay
* [ ] Skip back/forward ([app uses 15 seconds](https://github.com/bcc-code/bmm-app/blob/develop/BMM.Core/Implementations/Player/ViewModelHandlingMediaPlayerDecorator.cs#L22)) - _partially implemented for media-keys_
* [ ] Design
* [ ] Reorder queue (https://github.com/SortableJS/vue.draggable.next)

@sclausendk Should the progress of loading the track be visible to the use?

### CODE REVIEW

File: plugins/3.mediaPlayer.ts [hunk 1]
```diff
@@ -47,150 +15,11 @@ export default defineNuxtPlugin((nuxtApp) => {
...
-
-    activeMedia = new Audio(
-      authorizedUrl(track.media?.[0]?.files?.[0]?.url || "", authToken.value)
-    );
-    activeMedia.autoplay = true;
-    loading.value = true;
-    currentTrack.value = track;
-    paused.value = true;
-    ended.value = false;
-    useNuxtApp().$appInsights.trackEvent({
```
[the remark this answers is not in the document]
Comment (SimonSimCity) [reply plugins/3.mediaPlayer.ts:90]: They're in - just differently 😊 https://github.com/bcc-code/bmm-web/pull/178/files#diff-fee6f3bfb23303ee814a110258b4e1c994a519aa10d28b43d578763d27521d47R81-R108

File: pages/index.vue [hunk 2]
```diff
@@ -117,7 +120,7 @@ watch(
               :track="item"
               :is-track-type-known="true"
               show-thumbnail
-              @play-track="setCurrentTrack(item)"
+              @play-track="() => { const items = group.items.filter((c) : c is TrackModel => c.type === 'track'); setQueue(items, items.findIndex(track => track.id === item.id)) }"
```
Comment (sifferhans) [inline pages/index.vue:123]: Should we extract this to its own function? An inline function this large is not particularly readable in my opinion

**Review by sifferhans — APPROVED** [review]

**Review by sifferhans — APPROVED** [review]

**Comment by SimonSimCity** [comment]
Finished the last outstanding remainders - ready2merge. Once merged, all open points should be transferred into separate issues.

**Comment by kkuepper** [comment]
It's out-of-date with base branch. Otherwise I would have merged it

**Review by kkuepper — APPROVED** [review]

**Pull request merged**